openassessment/xblock/static/js/spec/lms/oa_base.js

```diff
@@ -50,21 +50,17 @@ describe("OpenAssessment.BaseView", function() {
     Args:
         callback (function): Function that takes no arguments.
     **/
-    var loadSubviews = function(callback) {
-        runs(function() {
-            view.load();
-        });
-
-        waitsFor(function() {
-            return !$(".openassessment__steps__step").hasClass('is--loading');
-        });
-
-        runs(function() {
-            return callback();
-        });
+    var loadSubviews = function(done) {
+        view.load();
+
+        // Wait for the is--loading class to have been removed
+        setTimeout(function() {
+            expect($(".openassessment__steps__step").hasClass('is--loading')).toBeFalsy();
+            done(); // call this to finish off the it block
+        }, 500);
```

Comment:
Good question. I thought it did the latter, but now I'm not so sure. I took this from here:

http://stackoverflow.com/questions/20963038/jasmine-2-0-refactoring-out-1-3s-runs-and-waitsfor

I really need to better understand what this code is doing. I'm not sure if it even needs to wait since everything should be happening synchronously. I'll look into it again.